drivers/gpio/gpio_cy8c95xx.c

```diff
@@ -0,0 +1,304 @@
+/*
+ * Copyright (c) 2021 Synopsys
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT cypress_cy8c95xx_gpio_port
+
+#include <errno.h>
+
+#include <kernel.h>
+#include <device.h>
+#include <init.h>
+#include <drivers/gpio.h>
+#include <drivers/i2c.h>
+#include <sys/byteorder.h>
+#include <sys/util.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(cy8c95xx, CONFIG_GPIO_LOG_LEVEL);
+
+#include "gpio_utils.h"
+
+/** Cache of the output configuration and data of the pins. */
+struct cy8c95xx_pin_state {
+	uint8_t dir;
+	uint8_t data;
+	uint8_t pullup;
+};
+
+/** Runtime driver data */
+struct cy8c95xx_drv_data {
+	/* gpio_driver_data needs to be first */
+	struct gpio_driver_data common;
+	const struct device *i2c_master;
```

Comment:
@galak should we require that this is moved to the const driver config and resolved at build time via `DEVICE_DT_GET`?

Reply:
It looks like we should use the new `DEVICE_DT_GET` macro to get handles of device driver dependencies. The macro allows us to get the handle at compile time. It's a more robust approach and has smaller footprint. This field can be moved to `struct cy8c95xx_config`. The `const char *i2c_master_dev_name;` in the other struct can be removed. There will also be no more need to call `device_get_binding` in the init function.

You can grep for `DEVICE_DT_GET` in `drivers/` folder for usage examples.